src/App.js

```diff
@@ -482,18 +499,44 @@ export const App = () => {
 	};
 
 	const onConnectWallet = async () => {
-		try {
-			const { uri, approval } = await wcSdk.createConnection(`neo3:${activeNet}`, ['invokeFunction', 'testInvoke', 'signMessage', 'verifyMessage']);
-			onModal('connectWallet', uri);
-			const session = await approval();
-			wcSdk.setSession(session);
-		} catch (error) {
-			onModal('failed', 'Something went wrong, contact the application administrator');
+		if (dapi) {
+			const provider = await dapi.getProvider();
+			const networks = await dapi.getNetworks();
+			const account = await dapi.getAccount();
+
+			setWalletData({
+				name: provider.name,
+				type: 'neo3',
+				net: networks.defaultNetwork.toLowerCase(),
+				account: account,
+				tokens: {
+					container: {},
+					object: null,
+				}
+			});
+			onPopup('success', 'Wallet connected');
+			onModal();
+
+			if (location.pathname.indexOf('/profile') === -1) {
+				document.location.href = "/profile";
+			}
+		} else {
+			try {
+				const { uri, approval } = await wcSdk.createConnection(`neo3:${activeNet}`, ['invokeFunction', 'testInvoke', 'signMessage', 'verifyMessage']);
+				onModal('connectWallet', uri);
+				const session = await approval();
+				wcSdk.setSession(session);
+			} catch (error) {
+				onModal('failed', 'Something went wrong, contact the application administrator');
+			}
 		}
 	}
 
 	const onDisconnectWallet = async () => {
-		await wcSdk.disconnect();
+		if (!dapi) {
```

Comment:
But what about `if dapi`? How can we disconnect?

Reply:
There is no such method in the [documentation](https://github.com/neo-ngd/neo-dapi-monorepo/tree/master/packages/neo-dapi), so we just redirect the user to the main page